test/Reader/FileReaderEncodingTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+use CsvParser\Parser;
+
+class FileReaderEncodingTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGermanBusinesses()
+    {
+        $parser = new Parser();
+        $csv = $parser->fromFile(__DIR__ . '/data/encoding/german_business.csv');
+        $rows = $parser->toArray($csv);
+
+        $this->assertEquals(3, count($rows));
+        $this->assertEquals('KFZ Meisterbetrieb Hönsch e.K.', $rows[0]['Firma']);
+        $this->assertEquals('Römerweg 8', $rows[1]['Straße']);
+        $this->assertEquals('München', $rows[2]['Stadt']);
+        $this->assertEquals('0211-123456', $rows[0]['Telefon']);
+        $this->assertEquals('service@müller-brot.de', $rows[2]['Email']);
+    }
+
+    public function testIso88591Encoding()
+    {
+        $parser = new Parser();
+        $csv = $parser->fromFile(__DIR__ . '/data/encoding/iso88591.csv');
+        $rows = $parser->toArray($csv);
+
+        $this->assertEquals(3, count($rows));
+        $this->assertEquals('Müller', $rows[0]['Name']);
+        $this->assertEquals('Café München', $rows[0]['Company']);
+        $this->assertEquals('Düsseldorf', $rows[0]['City']);
+        $this->assertEquals('Höör AB', $rows[1]['Company']);
+        $this->assertEquals('Göteborg', $rows[1]['City']);
+        $this->assertEquals('Schöne Grüße GmbH', $rows[2]['Company']);
+    }
+
+    public function testIso885915Encoding()
+    {
+        $parser = new Parser();
+        $csv = $parser->fromFile(__DIR__ . '/data/encoding/iso885915.csv');
+        $rows = $parser->toArray($csv);
+
+        $this->assertEquals(2, count($rows));
+        $this->assertEquals('Product A', $rows[0]['Name']);
+        // Note: Euro symbol in ISO-8859-15 may be converted to ¤ if detected as ISO-8859-1
+        $this->assertStringContainsString('25.50', $rows[0]['Price']);
+        $this->assertEquals('Européen', $rows[0]['Notes']);
+        $this->assertStringContainsString('30.00', $rows[1]['Price']);
+        $this->assertEquals('Français', $rows[1]['Notes']);
+    }
+
+    public function testWindows1252Encoding()
+    {
+        $parser = new Parser();
+        $csv = $parser->fromFile(__DIR__ . '/data/encoding/windows1252.csv');
+        $rows = $parser->toArray($csv);
+
+        $this->assertEquals(2, count($rows));
+        $this->assertEquals('Smith', $rows[0]['Name']);
+        $this->assertStringContainsString('dash', $rows[0]['Notes']);
+        $this->assertEquals('García', $rows[1]['Name']);
+        $this->assertEquals('López & Hernández', $rows[1]['Company']);
+        // Trademark and Copyright symbols should be preserved
+        $this->assertStringContainsString('Copyright', $rows[1]['Notes']);
```

Comment:
the comment above says copyright symbols should be preserved but the test seems to only check for the text copyright, is that correct?